include/dmlc/optional.h

```diff
@@ -173,9 +176,49 @@ std::istream &operator>>(std::istream &is, optional<T> &t) {
   }
   return is;
 }
+/*! \brief specialization of '>>' istream parsing for optional<bool>
+ *
+ * Permits use of generic parameter FieldEntry<DType> class to create
+ * FieldEntry<optional<bool>> without explicit specialization.
+ *
+ *  \code
+ *    dmlc::optional<bool> x;
+ *    std::string s1 = "true";
+ *    std::istringstream is1(s1);
+ *    s1 >> x;  // x == optional<bool>(true)
+ *
+ *    std::string s2 = "None";
+ *    std::istringstream is2(s2);
+ *    s2 >> x;  // x == optional<bool>()
+ *  \endcode
+ *
+ *  \param is input stream
+ *  \param t target optional<bool> object
+ *  \return input stream
+ */
+inline std::istream &operator>>(std::istream &is, optional<bool> &t) {
+  std::string s;
+  is >> s;
```

Comment:
I've fixed the istream reader to pass on your posed nnvm::Tuple testcase.  I can contribute a completed version of the test to nnvm after this dmlc-core PR is accepted, if you wish.